src/shims/windows/handle.rs

```diff
@@ -105,6 +117,9 @@ impl Handle {
             Self::NULL_DISCRIMINANT if data == 0 => Some(Self::Null),
             Self::PSEUDO_DISCRIMINANT => Some(Self::Pseudo(PseudoHandle::from_value(data)?)),
             Self::THREAD_DISCRIMINANT => Some(Self::Thread(ThreadId::new_unchecked(data))),
+            #[expect(clippy::cast_possible_wrap)]
+            Self::FILE_DISCRIMINANT => Some(Self::File(data as i32)),
```

Comment:
I wish we had some nice way to assert that this cast is not truncating. In lieu of that, maybe let's do this?
```suggestion
            Self::FILE_DISCRIMINANT => {
                // This cast preserves all bits.
                assert_eq!(size_of_val(&data), size_of::<i32>());
                Some(Self::File(data as i32))
            }
```

Reply:
You resolved this but didn't actually change the code?

Reply:
Ah you did, I just missed the commit. :)